server/core/include/administration_utilities.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef IRODS_ADMINISTRATION_UTILITIES_HPP
+#define IRODS_ADMINISTRATION_UTILITIES_HPP
+
+#include <string_view>
+
+struct RsComm;
+
+namespace irods
+{
+    /// \brief Create a user with the given name, type, and auth string
+    ///
+    /// \parblock
+    /// If _user_name does not contain the zone name using the octothorpe delimiter, the
+    /// _zone_name will be used. Both of these are available here because the GeneralAdmin
+    /// interface is flexible enough to allow the zone name to be passed through the username
+    /// string or through the generalAdninInp struct.
```

Comment:
Is there a test where the zone name is both *in* the username string *and* the _zone_name?   And if they don't agree... seems the _user_name string wins?

Reply:
Looking at the implementation again, we actually aren't using _zone_name at all. And `iadmin` is passing an empty string for that parameter. So I'm just going to ignore it entirely and look for the zone name in the user name.